packages/hooks/use-theme/src/index.ts

```diff
@@ -0,0 +1,81 @@
+import {useCallback, useEffect, useState} from "react";
+
+// constant properties for Theme
+export const ThemeProps = {
+  // localStorage key for storing the current theme
+  KEY: "nextui-theme",
+  // light theme
+  LIGHT: "light",
+  // dark theme
+  DARK: "dark",
+  // system theme
+  SYSTEM: "system",
+} as const;
+
+// type definition for Theme using system theme names or custom theme names
+export type customTheme = string;
+export type Theme =
+  | typeof ThemeProps.LIGHT
+  | typeof ThemeProps.DARK
+  | typeof ThemeProps.SYSTEM
+  | customTheme;
+
+/**
+ * React hook to switch between themes
+ *
+ * @param defaultTheme the default theme name (e.g. light, dark, purple-dark and etc)
+ * @returns An object containing the current theme and theme manipulation functions
+ */
+export function useTheme(defaultTheme?: Theme) {
+  const [theme, setThemeState] = useState<Theme>(() => {
```

Comment:
It would be more effective to initialize the state with either the value from localStorage or the `defaultTheme`. Additionally, it makes sense to create an `applyTheme` function to handle theme changes when they occur. You can refer to the following implementation.
https://github.com/pacocoursey/next-themes/blob/main/next-themes/src/index.tsx#L42-L49

```suggestion
export function useTheme(defaultTheme: Theme = ThemeProps.SYSTEM) {
  const [theme, setThemeState] = useState<Theme>(() => localStorage.getItem(ThemeProps.KEY) ?? defaultTheme);
```

Reply:
I think there is no need to modify line 30, the version before modification is safer and more complete, just need to change line 29.